pgmpy/models/MarkovChain.py

```diff
@@ -363,3 +363,37 @@ def random_state(self):
         [State('diff', 2), State('intel', 1)]
         """
         return [State(var, np.random.randint(self.cardinalities[var])) for var in self.variables]
+
+    def copy(self):
+        """
+        Returns copy of Markov Chain
+
+        Return Type:
+        ------------
+        MarkovChain : Copy of MarkovChain
+
+        Examples:
+        ---------
+        >>> from pgmpy.models.MarkovChain import MarkovChain
+        >>> from pgmpy.factors import State
+        >>> model = MarkovChain()
+        >>> model.add_variables_from(['intel', 'diff'], [3, 2])
+        >>> intel_tm = {0: {0: 0.2, 1: 0.4, 2:0.4}, 1: {0: 0, 1: 0.5, 2: 0.5}, 2: {0: 0.3, 1: 0.3, 2: 0.4}}
+        >>> model.add_transition_model('intel', intel_tm)
+        >>> diff_tm = {0: {0: 0.5, 1: 0.5}, 1: {0: 0.25, 1:0.75}}
+        >>> model.add_transition_model('diff', diff_tm)
+        >>> from pgmpy.factors import State
+        >>> model.set_start_state([State('intel', 0), State('diff', 2)])
+        >>> model_copy = model.copy()
+        """
+        variable_copy = []
+        cardinalities_copy = []
+        for var,card in self.cardinalities.items():
+            variable_copy.append(var)
+            cardinalities_copy.append(card)
+
+        copy = MarkovChain(variables=variable_copy, card=cardinalities_copy, start_state=self.state)
+        if self.transition_models:
+            copy.transition_models = self.transition_models
```

Comment:
@anchitjain1234 No. Since those are `dict` you can simply use `copy` method of `dict`